src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,161 @@
+  const [loading, setLoading] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const filteredPeople = useMemo(() => {
+    return people
+      .filter(person => person?.name?.toLowerCase()
+        .includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery, people]);
+
+  const delayQuery = useMemo(() => debounce((value: string) => {
+    setAppliedQuery(value.trim());
+  }, delay), [delay]);
+
+  const handleInputChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const value = event.target.value.trim();
+
+      setQuery(value);
+      setTitle('No selected person');
+      setLoading(true);
+      delayQuery(value);
+    }, [delayQuery],
+  );
+
+  const handleOnBlur = useCallback(() => {
+    setTimeout(() => {
+      if (dropdownRef.current
+        && !dropdownRef.current.contains(document.activeElement as Node)) {
+        setIsVisible(false);
+        setLoading(false);
+      }
+    }, 100);
+  }, []);
+
+  const handlePersonSelect = useCallback((person: Person | null) => {
+    if (person) {
+      setQuery(person.name);
+      setTitle(`${person.name} (${person.born} - ${person.died})`);
+    } else {
+      setQuery('');
+      setTitle('No selected person');
+    }
+
+    setIsVisible(false);
+    setLoading(false);
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      delayQuery.cancel();
+    };
+  }, [delayQuery]);
+
+  useEffect(() => {
+    const fetchData = () => {
+      setAppliedQuery(query);
+      setLoading(true);
+      setTimeout(() => {
+        setLoading(false);
+      }, 1000);
+    };
+
+    fetchData();
+  }, [query]);
+
+  return (
+    <>
+      <h1 className="title" data-cy="title">
+        {title}
+      </h1>
+
+      <div className="dropdown is-active" ref={dropdownRef}>
+        <div className="dropdown-trigger">
+          <input
+            type="text"
+            placeholder="Enter a part of the name"
+            className="input"
+            data-cy="search-input"
+            value={query}
+            onChange={handleInputChange}
+            onFocus={() => setIsVisible(true)}
+            onBlur={handleOnBlur}
+          />
+        </div>
+
+        {isVisible && !loading && (
+          <div
+            className="dropdown-menu"
+            role="menu"
+            data-cy="suggestions-list"
+          >
+            <div className="dropdown-content">
+              {filteredPeople.length === 0 ? (
+                <div
+                  className="
+            notification
+            is-danger
+            is-light
+            mt-3
+            is-align-self-flex-start
+          "
+                  role="alert"
+                  data-cy="no-suggestions-message"
+                >
+                  <p className="has-text-danger">No matching suggestions</p>
+                </div>
+              ) : (
+                filteredPeople.map((person) => (
+                  <div
```

Comment:
Consider will be better if you use the semantic tag for list of people

Reply:
not sure that I understood exactly how it should be changed, wrapped in `ul`